src/NodeApi/JSValueScope.cs

```diff
@@ -82,25 +82,50 @@ public sealed class JSValueScope : IDisposable
     /// <summary>
     /// Gets the current JS value scope.
     /// </summary>
-    /// <exception cref="InvalidOperationException">No scope was established for the current
+    /// <exception cref="JSInvalidScopeException">No scope was established for the current
     /// thread.</exception>
     public static JSValueScope Current => s_currentScope ??
-        throw new InvalidOperationException("No current scope.");
+        throw new JSInvalidScopeException(currentScope: null);
+
+    internal int ThreadId { get; }
 
     public bool IsDisposed { get; private set; }
```

Comment:
Where do we check that the scope is disposed? It must make any related JSValue invalid.

Reply:
Getting either the `napi_value` or `napi_env` now checks for disposed scope.